src/System.IO/tests/BinaryWriter/BinaryWriter.WriteTests.cs

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using System;
+using System.IO;
+using System.Text;
+
+namespace System.IO.Tests
+{
+    public class BinaryWriter_WriteTests
+    {
+        protected virtual Stream CreateStream()
+        {
+            return new MemoryStream();
+        }
+
+        [Fact]
+        public void BinaryWriter_WriteBoolTest()
+        {
+            // [] Write a series of booleans to a stream
+            using(Stream mstr = CreateStream())
+            using(BinaryWriter dw2 = new BinaryWriter(mstr))
+            using(BinaryReader dr2 = new BinaryReader(mstr))
+            {
+                dw2.Write(false);
+                dw2.Write(false);
+                dw2.Write(true);
+                dw2.Write(false);
+                dw2.Write(true);
+                dw2.Write(5);
+                dw2.Write(0);
+
+                dw2.Flush();
+                mstr.Position = 0;
+
+                Assert.False(dr2.ReadBoolean()); //false
+                Assert.False(dr2.ReadBoolean()); //false
+                Assert.True(dr2.ReadBoolean());  //true
+                Assert.False(dr2.ReadBoolean()); //false
+                Assert.True(dr2.ReadBoolean());  //true
+                Assert.True(dr2.ReadBoolean());  //5
+                Assert.False(dr2.ReadBoolean()); //0
```

Comment:
Yeah, looks like it works but for the wrong reason. I'll fix it to read an int instead, though the test is named "writebool" so I'm not sure why it's writing integers at all.